### DISCUSSION TITLE
Clearance Height

### DISCUSSION THREAD

**Discussion by Sstaceyh74** [winder/Universal-G-Code-Sender#1753] [Help] [answered]
Hi all, hope everyone is well!

Is there a setting for the clearance height, I ask as when I hit the "Return to Zero" bottom the head drops, moved to the X & Y home then raises, I would have thought that it would go up move then drop??

If there is a clearance height maybe I have a negative number in it... or it need to be a negative etc...

Stacey

**Comment by breiler** [comment, marked as answer]
The clearance height should be set to positive values in these settings:
![image](https://user-images.githubusercontent.com/8962024/147445951-83785274-8a1c-4ed7-b0d1-a0724c8eed13.png)

The behaviour should be like this psuedo code ([source](https://github.com/winder/Universal-G-Code-Sender/blob/master/ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java#L163))
```
ClearanceHeight = 5mm

If Z < ClearanceHeight then move Z to ClearanceHeight
Move XY to 0
Move Z to 0
```

Are your machine coordinate system set up like this?
![[image](https://www.autodesk.com/products/fusion-360/blog/cnc-coordinate-system-made-easy/)](https://user-images.githubusercontent.com/8962024/147446892-0e82bf6d-9077-4178-a45c-9a46e050ba3a.png)
